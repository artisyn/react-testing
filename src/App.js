import React, { useEffect, useState } from 'react';
import './App.css';

function App() {
	const [data, setData] = useState(null);
	const [toggle, setToggle] = useState(false);
	const [value, setValue] = useState('');

	const onClick = () => setToggle((prev) => !prev);

	useEffect(() => {
		setTimeout(() => {
			setData({});
		}, 1000);
	}, []);

	return (
		<div className="App">
			{toggle === true && <div data-testid="hhh"> toggle is true</div>}
			<h1 data-testid="datah1">{value}</h1>
			<h1>Hello World!</h1>
			{data && <div style={{ color: 'red' }}>data</div>}
			<button data-testid="toggle-btn" onClick={onClick}>
				Click me!
			</button>
			<input
				value={value}
				onChange={(e) => {
					setValue(e.target.value);
				}}
				type={'text'}
				placeholder="type some text here..."
			/>
		</div>
	);
}

export default App;
