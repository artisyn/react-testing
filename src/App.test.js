import { render, screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from './App';

describe('test app', () => {
	// test('renders learn react link', () => {
	// 	render(<App />);
	// 	const helloWorld = screen.getByText(/hello world/i);
	// 	expect(helloWorld).toBeInTheDocument();
	// 	const button = screen.getByRole('button');
	// 	expect(button).toBeInTheDocument();
	// 	const input = screen.getByPlaceholderText(/type some text here.../i);
	// 	expect(input).toMatchSnapshot();
	// 	screen.debug();
	// });
	// test('val', () => {
	// 	render(<App />);
	// 	const helloWorld2 = screen.queryByText(/hello2/i);
	// 	expect(helloWorld2).toBeNull();
	// });
	// test('async', async () => {
	// 	render(<App />);
	// 	screen.debug();
	// 	const asyncData = await screen.findByText(/data/i);
	// 	expect(asyncData).toBeInTheDocument();
	// 	expect(asyncData).toHaveStyle({ color: 'red' });
	// 	screen.debug();
	// });
	// test('click event', () => {
	// 	render(<App />);
	// 	screen.debug();
	// 	const btn = screen.getByTestId('toggle-btn');
	// 	expect(screen.queryByTestId('hhh')).toBeNull();
	// 	fireEvent.click(btn);
	// 	screen.debug();
	// 	expect(screen.queryByTestId('hhh')).toBeInTheDocument();
	// });
	test('input event', () => {
		render(<App />);

		const input = screen.getByPlaceholderText(/type some text here.../i);
		const btn = screen.getByTestId('toggle-btn');
		expect(screen.queryByTestId('datah1')).toContainHTML('');
		// fireEvent.input(input, {
		// 	target: { value: '123123' },
		// });
		userEvent.type(input, '123123');

		expect(screen.queryByTestId('datah1')).toContainHTML('123123');
	});
});
