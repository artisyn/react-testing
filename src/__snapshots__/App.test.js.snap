// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<input
  placeholder="type some text here..."
  type="text"
/>
`;

exports[`test app renders learn react link 1`] = `
<input
  placeholder="type some text here..."
  type="text"
/>
`;
